src/api/api.c

```diff
@@ -849,8 +849,29 @@ void getAllQueries(const char *client_message, const int *sock)
 			continue;
 
 		// Skip if domain is not identical with what the user wants to see
-		if(filterdomainname && query->domainID != domainid)
-			continue;
+		if(filterdomainname)
+		{
+			// Check direct match
+			if(query->domainID == domainid)
```

Comment:
Just so I know I've got my head around this, does this also read as: (might be slightly syntactically wrong!)

```c
if (! ((query->status == QUERY_GRAVITY_CNAME || query->status == QUERY_BLACKLIST_CNAME || query->status == QUERY_REGEX_CNAME) && query ->CNAME_domainID == domainid)) && ! (query->domainID === domainid))
{
 continue;
}
```

Reply:
Yes, but I thought it's easier to have it is smaller pieces. I already checked in the assembly output that the compiler optimizer is cleaning up my slightly more verbose syntax by doing some very fancy optimized branching so the more explicit writing style doesn't seem to cause any costs.